### PR TITLE
geom_density() as filled area chart

### PR DESCRIPTION
**Pull request by cpsievert**
See #199.

### CODE REVIEW

**Comment by cpsievert** [comment]
Note that 261244ada631b2dcf9d603989572621c65445160 is really the only commit relevant to this PR (the rest of the commits are carried over from #200)

**Comment by cpsievert** [comment]
New tests --

http://ropensci.github.io/plotly-test-table/tables/4659669a2ea26bb8537cfc70b6e714afd6f62094/density-simple.html

http://ropensci.github.io/plotly-test-table/tables/4659669a2ea26bb8537cfc70b6e714afd6f62094/density-fill.html

http://ropensci.github.io/plotly-test-table/tables/4659669a2ea26bb8537cfc70b6e714afd6f62094/density-color.html

Something strange is happening with the density-fill example. If I click on "fork and edit", it takes me to another page that renders the plot correctly 

screen shot 2015-04-13 at 4 50 33 pm

Could this be some sort of bug with plotly itself @chriddyp @mkcor ?

**Comment by tdhock** [comment]
Nice tests, and yeah that it weird about the difference between the rendered plotlys.

@alexcjohnson do you have any idea why this same plotly is rendered differently in these two cases?

**Comment by cpsievert** [comment]
I don't think I can do anything to fix the problem above, so I'd prefer to merge, then post that rendering issue somewhere more appropriate. Any ideas on where that might be @chriddyp @mkcor @tdhock?

Please review and +1 when you get a chance.

**Comment by alexcjohnson** [comment]
Ah, apologies for not looking at this before. I just did tried "fork and edit" and didn't see a change... but I can imagine where it could come from. The issue is the `null` at the end of the x array in the first trace. You're right that this *shouldn't* break it like this, so I'll open an issue for it on the plotly side - but I also imagine it's not quite what you had in mind here...

**Comment by cpsievert** [comment]
Nice, thanks for catching this @alexcjohnson!

I'll push a fix shortly.

**Comment by cpsievert** [comment]
Everything looks good now!

http://ropensci.github.io/plotly-test-table/tables/4a336c65d8cdc0a4f974912a4e93d45e1852e003/index.html

File: tests/testthat/test-ggplot-density.R [hunk 1]
```diff
@@ -1,15 +1,45 @@
 context("Probability density")
 
+expect_traces <- function(gg, n.traces, name){
```
Comment (mkcor) [inline tests/testthat/test-ggplot-density.R]: Space after closing `)`.

**Comment by mkcor** [comment]
Nice work, @cpsievert !
I only spotted a missing whitespace. Up to that, that's a +1 from me.

**Pull request merged**